#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/animals/protos/Dog.proto"
EXTERNPROTO "../protos/RobocupGoal.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/balls/protos/RobocupSoccerBall.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/factory/containers/protos/WoodenBox.proto"

WorldInfo {
}
Viewpoint {
  orientation 0.009795535481006175 0.998345059850636 -0.056667353533321765 0.34260511723955256
  position -4.9029578789400325 2.2583209944569256 1.111325747108285
}
TexturedBackground {
}
Robot {
  translation -3.03643 3.34848 0.0440903
  rotation -0.9967827458666172 -0.02243977925921866 0.07694552520715238 0.01693702183766641
  children [
    DEF left_middle_dist DistanceSensor {
      translation 0.02 -0.041 -0.03
      rotation 0.577349935856137 0.5773509358560258 -0.577349935856137 2.09439
      name "left_middle_dist"
      type "infra-red"
    }
    DEF right_middle_dist DistanceSensor {
      translation -0.0219043 -0.038 -0.03
      rotation 0.5199877851067723 0.6776617199455095 -0.5199877851067723 1.95044
      name "right_middle_dist"
      type "infra-red"
    }
    DEF green_camera Camera {
      translation 0.0378109 -0.0395095 -0.00965926
      rotation 0.3424559939220911 0.8267659853265575 -0.4462979920791033 1.49626
      name "g_camera"
      fieldOfView 0.1
      width 30
      height 30
    }
    Gyro {
    }
    DEF penalty_distance_sensor DistanceSensor {
      translation 4.07848e-08 -0.04 0.03
      rotation 3.69431e-07 9.05875e-07 1 -1.5707953071795862
      name "p_distance"
      lookupTable [
        0 0 0
        0.25 1000 0
      ]
    }
    DEF penalty_canmera Camera {
      translation 0.00912765 -0.0393306 -0.02
      rotation 0.2452438906720146 0.31960885752064033 -0.9152625919824343 1.3954
      name "p_camera"
      fieldOfView 0.1
      width 30
      height 30
    }
    DEF left_obstacle DistanceSensor {
      translation 0.06 0 0
      name "left_ob"
    }
    DEF right_obstacle DistanceSensor {
      translation -0.06 0 0
      rotation 0 0 1 3.14159
      name "right_ob"
    }
    DEF up_right DistanceSensor {
      translation -0.0430743 -0.0364198 0.000326735
      rotation -0.22086490538558085 0.9562925903420335 -0.19163291790802078 1.87491
      name "up_right_li"
      type "infra-red"
    }
    Lidar {
      translation -0.065 -2.06647e-15 0
      rotation 0 0 1 3.14159
      name "lidar(2)"
      horizontalResolution 20
      numberOfLayers 1
    }
    Lidar {
      translation 0.065 0 0
      rotation 0 0 -1 -0.2618053071795865
      name "lidar(1)"
      horizontalResolution 20
      numberOfLayers 1
    }
    DEF line_right DistanceSensor {
      translation -0.04 -0.04 -0.0299998
      rotation 0 1 0 1.5708
      name "li_r"
      type "infra-red"
    }
    DEF line_middle DistanceSensor {
      translation -1.10196e-07 -0.04 -0.03
      rotation 0 1 0 1.5708
      name "li_m"
      type "infra-red"
    }
    DEF line_left DistanceSensor {
      translation 0.04 -0.04 -0.0300001
      rotation 0 1 0 1.5708
      name "li_l"
      type "infra-red"
    }
    Lidar {
      translation 4.07848e-08 -0.04 0.03
      rotation 0 0 1 -1.5707953071795862
      horizontalResolution 20
      numberOfLayers 1
    }
    DEF left_eye Solid {
      translation 0.02 -0.04 0.02
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0 0 0
            }
          }
          geometry Sphere {
            radius 0.01
          }
        }
      ]
      name "left_eye"
    }
    DEF right_eye Solid {
      translation -0.02 -0.04 0.02
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0 0 0
            }
          }
          geometry Sphere {
            radius 0.01
          }
        }
      ]
      name "right_eye"
    }
    DEF left_wheel HingeJoint {
      jointParameters HingeJointParameters {
        position 5398.710541330062
        anchor 0.055 0 0
      }
      device [
        RotationalMotor {
          name "left_motor"
          maxVelocity 100
        }
      ]
      endPoint Solid {
        translation 0.055177850978196274 3.235313871749245e-05 -6.466068570916105e-06
        rotation -0.5535446794427326 -0.6238075113114788 -0.5517721238808841 4.24914221935053
        children [
          DEF wl Shape {
            appearance Appearance {
              material Material {
                diffuseColor 1 1 0
                emissiveColor 1 0.333333 0
              }
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        boundingObject USE wl
        physics Physics {
        }
      }
    }
    DEF right_wheel HingeJoint {
      jointParameters HingeJointParameters {
        position 5550.888797634383
        anchor -0.055 0 0
      }
      device [
        RotationalMotor {
          name "right_motor"
          maxVelocity 100
        }
      ]
      endPoint Solid {
        translation -0.055289410310266796 2.6285977408946337e-05 2.938766267521209e-05
        rotation 0.7015011364905448 0.10887505858506279 0.7043027595576872 2.927410038506952
        children [
          DEF wr Shape {
            appearance Appearance {
              material Material {
                diffuseColor 1 1 0
                emissiveColor 1 0 0
              }
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        name "solid(1)"
        boundingObject USE wr
        physics Physics {
        }
      }
    }
    DEF body Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0.333333 0 0.498039
          emissiveColor 0.333333 0 0.498039
        }
      }
      geometry Box {
        size 0.09 0.08 0.078
      }
    }
  ]
  boundingObject USE body
  physics Physics {
  }
  controller "main_c"
}
TexturedBackgroundLight {
}
Solid {
  translation -0.01 0 0
  children [
    Solid {
      children [
        Shape {
          appearance PBRAppearance {
            baseColorMap ImageTexture {
              url [
                "../main.png"
              ]
            }
            roughness 1
            roughnessMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/textures/parquetry/chequered_parquetry_roughness.jpg"
              ]
            }
            metalness 0
            normalMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/textures/parquetry/chequered_parquetry_normal.jpg"
              ]
            }
            occlusionMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/textures/parquetry/chequered_parquetry_occlusion.jpg"
              ]
            }
          }
          geometry IndexedFaceSet {
            coord Coordinate {
              point [
                -3.5 -3.5 0
                3.5 -3.5 0
                -3.5 3.5 0
                3.5 3.5 0
              ]
            }
            texCoord TextureCoordinate {
              point [
                0 0
                1 0
                0 1
                1 1
              ]
            }
            coordIndex [
              2, 0, 1, 3, -1
            ]
            texCoordIndex [
              2, 0, 1, 3, -1
            ]
          }
        }
      ]
      name "floor"
      model "floor"
      boundingObject Plane {
        size 7 7
      }
      locked TRUE
    }
    Solid {
      translation 3.505 0 0.05
      children [
        Shape {
          appearance PBRAppearance {
            baseColorMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/textures/brushed_aluminium/brushed_aluminium_base_color.jpg"
              ]
            }
            roughnessMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/textures/brushed_aluminium/brushed_aluminium_roughness.jpg"
              ]
            }
            normalMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/textures/brushed_aluminium/brushed_aluminium_normal.jpg"
              ]
            }
          }
          geometry IndexedFaceSet {
            coord Coordinate {
              point [
                -0.005 3.5 0.05
                0.005 3.5 0.05
                0.005 3.5 -0.05
                -0.005 3.5 -0.05
                -0.005 -3.5 -0.05
                0.005 -3.5 -0.05
                0.005 -3.5 0.05
                -0.005 -3.5 0.05
                -0.005 -3.5 0.05
                0.005 -3.5 0.05
                0.005 3.5 0.05
                -0.005 3.5 0.05
                0.005 -3.5 -0.05
                -0.005 -3.5 -0.05
                -0.005 3.5 -0.05
                0.005 3.5 -0.05
                -0.005 -3.5 -0.05
                -0.005 -3.5 0.05
                -0.005 3.5 0.05
                -0.005 3.5 -0.05
                0.005 -3.5 0.05
                0.005 -3.5 -0.05
                0.005 3.5 -0.05
                0.005 3.5 0.05
              ]
            }
            texCoord TextureCoordinate {
              point [
                0.0014285714285714286 0.014285714285714287
                0 0.014285714285714287
                0 0
                0.0014285714285714286 0
                0 0
                0.0014285714285714286 0
                0.0014285714285714286 0.014285714285714287
                0 0.014285714285714287
                0 0.0014285714285714286
                0 0
                1 0
                1 0.0014285714285714286
                0 0.0014285714285714286
                0 0
                1 0
                1 0.0014285714285714286
                1 0
                1 0.014285714285714287
                0 0.014285714285714287
                0 0
                0 0.014285714285714287
                0 0
                1 0
                1 0.014285714285714287
              ]
            }
            coordIndex [
              0, 1, 2, 3, -1, 4, 5, 6, 7, -1, 8, 9, 10, 11, -1, 12, 13, 14, 15, -1, 16, 17, 18, 19, -1, 20, 21, 22, 23, -1
            ]
            creaseAngle 1.5
          }
        }
      ]
      name "wall"
      model "box"
      boundingObject Box {
        size 0.01 7 0.1
      }
    }
    Solid {
      translation -3.505 0 0.05
      children [
        Shape {
          appearance PBRAppearance {
            baseColorMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/textures/brushed_aluminium/brushed_aluminium_base_color.jpg"
              ]
            }
            roughnessMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/textures/brushed_aluminium/brushed_aluminium_roughness.jpg"
              ]
            }
            normalMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/textures/brushed_aluminium/brushed_aluminium_normal.jpg"
              ]
            }
          }
          geometry IndexedFaceSet {
            coord Coordinate {
              point [
                -0.005 3.5 0.05
                0.005 3.5 0.05
                0.005 3.5 -0.05
                -0.005 3.5 -0.05
                -0.005 -3.5 -0.05
                0.005 -3.5 -0.05
                0.005 -3.5 0.05
                -0.005 -3.5 0.05
                -0.005 -3.5 0.05
                0.005 -3.5 0.05
                0.005 3.5 0.05
                -0.005 3.5 0.05
                0.005 -3.5 -0.05
                -0.005 -3.5 -0.05
                -0.005 3.5 -0.05
                0.005 3.5 -0.05
                -0.005 -3.5 -0.05
                -0.005 -3.5 0.05
                -0.005 3.5 0.05
                -0.005 3.5 -0.05
                0.005 -3.5 0.05
                0.005 -3.5 -0.05
                0.005 3.5 -0.05
                0.005 3.5 0.05
              ]
            }
            texCoord TextureCoordinate {
              point [
                0.0014285714285714286 0.014285714285714287
                0 0.014285714285714287
                0 0
                0.0014285714285714286 0
                0 0
                0.0014285714285714286 0
                0.0014285714285714286 0.014285714285714287
                0 0.014285714285714287
                0 0.0014285714285714286
                0 0
                1 0
                1 0.0014285714285714286
                0 0.0014285714285714286
                0 0
                1 0
                1 0.0014285714285714286
                1 0
                1 0.014285714285714287
                0 0.014285714285714287
                0 0
                0 0.014285714285714287
                0 0
                1 0
                1 0.014285714285714287
              ]
            }
            coordIndex [
              0, 1, 2, 3, -1, 4, 5, 6, 7, -1, 8, 9, 10, 11, -1, 12, 13, 14, 15, -1, 16, 17, 18, 19, -1, 20, 21, 22, 23, -1
            ]
            creaseAngle 1.5
          }
        }
      ]
      name "wall#2"
      model "box"
      boundingObject Box {
        size 0.01 7 0.1
      }
    }
    Solid {
      translation 0 3.505 0.05
      children [
        Shape {
          appearance PBRAppearance {
            baseColorMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/textures/brushed_aluminium/brushed_aluminium_base_color.jpg"
              ]
            }
            roughnessMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/textures/brushed_aluminium/brushed_aluminium_roughness.jpg"
              ]
            }
            normalMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/textures/brushed_aluminium/brushed_aluminium_normal.jpg"
              ]
            }
          }
          geometry IndexedFaceSet {
            coord Coordinate {
              point [
                -3.51 0.005 0.05
                3.51 0.005 0.05
                3.51 0.005 -0.05
                -3.51 0.005 -0.05
                -3.51 -0.005 -0.05
                3.51 -0.005 -0.05
                3.51 -0.005 0.05
                -3.51 -0.005 0.05
                -3.51 -0.005 0.05
                3.51 -0.005 0.05
                3.51 0.005 0.05
                -3.51 0.005 0.05
                3.51 -0.005 -0.05
                -3.51 -0.005 -0.05
                -3.51 0.005 -0.05
                3.51 0.005 -0.05
                -3.51 -0.005 -0.05
                -3.51 -0.005 0.05
                -3.51 0.005 0.05
                -3.51 0.005 -0.05
                3.51 -0.005 0.05
                3.51 -0.005 -0.05
                3.51 0.005 -0.05
                3.51 0.005 0.05
              ]
            }
            texCoord TextureCoordinate {
              point [
                1 0.014245014245014247
                0 0.014245014245014247
                0 0
                1 0
                0 0
                1 0
                1 0.014245014245014247
                0 0.014245014245014247
                0 1
                0 0
                0.0014245014245014246 0
                0.0014245014245014246 1
                0 1
                0 0
                0.0014245014245014246 0
                0.0014245014245014246 1
                0.0014245014245014246 0
                0.0014245014245014246 0.014245014245014247
                0 0.014245014245014247
                0 0
                0 0.014245014245014247
                0 0
                0.0014245014245014246 0
                0.0014245014245014246 0.014245014245014247
              ]
            }
            coordIndex [
              0, 1, 2, 3, -1, 4, 5, 6, 7, -1, 8, 9, 10, 11, -1, 12, 13, 14, 15, -1, 16, 17, 18, 19, -1, 20, 21, 22, 23, -1
            ]
            creaseAngle 1.5
          }
        }
      ]
      name "wall#3"
      model "box"
      boundingObject Box {
        size 7.02 0.01 0.1
      }
    }
    Solid {
      translation 0 -3.505 0.05
      children [
        Shape {
          appearance PBRAppearance {
            baseColorMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/textures/brushed_aluminium/brushed_aluminium_base_color.jpg"
              ]
            }
            roughnessMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/textures/brushed_aluminium/brushed_aluminium_roughness.jpg"
              ]
            }
            normalMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/textures/brushed_aluminium/brushed_aluminium_normal.jpg"
              ]
            }
          }
          geometry IndexedFaceSet {
            coord Coordinate {
              point [
                -3.51 0.005 0.05
                3.51 0.005 0.05
                3.51 0.005 -0.05
                -3.51 0.005 -0.05
                -3.51 -0.005 -0.05
                3.51 -0.005 -0.05
                3.51 -0.005 0.05
                -3.51 -0.005 0.05
                -3.51 -0.005 0.05
                3.51 -0.005 0.05
                3.51 0.005 0.05
                -3.51 0.005 0.05
                3.51 -0.005 -0.05
                -3.51 -0.005 -0.05
                -3.51 0.005 -0.05
                3.51 0.005 -0.05
                -3.51 -0.005 -0.05
                -3.51 -0.005 0.05
                -3.51 0.005 0.05
                -3.51 0.005 -0.05
                3.51 -0.005 0.05
                3.51 -0.005 -0.05
                3.51 0.005 -0.05
                3.51 0.005 0.05
              ]
            }
            texCoord TextureCoordinate {
              point [
                1 0.014245014245014247
                0 0.014245014245014247
                0 0
                1 0
                0 0
                1 0
                1 0.014245014245014247
                0 0.014245014245014247
                0 1
                0 0
                0.0014245014245014246 0
                0.0014245014245014246 1
                0 1
                0 0
                0.0014245014245014246 0
                0.0014245014245014246 1
                0.0014245014245014246 0
                0.0014245014245014246 0.014245014245014247
                0 0.014245014245014247
                0 0
                0 0.014245014245014247
                0 0
                0.0014245014245014246 0
                0.0014245014245014246 0.014245014245014247
              ]
            }
            coordIndex [
              0, 1, 2, 3, -1, 4, 5, 6, 7, -1, 8, 9, 10, 11, -1, 12, 13, 14, 15, -1, 16, 17, 18, 19, -1, 20, 21, 22, 23, -1
            ]
            creaseAngle 1.5
          }
        }
      ]
      name "wall#4"
      model "box"
      boundingObject Box {
        size 7.02 0.01 0.1
      }
    }
  ]
  name "rectangle arena"
  model "rectangle arena"
}
WoodenBox {
  translation 1.45205 3.01314 0.04803800000000001
  rotation -0.9999999999612821 -6.222381127979863e-06 6.222381128010689e-06 1.5707963268336147
  size 0.1 0.1 0.1
  mass 5
}
WoodenBox {
  translation -0.30331099991574567 -2.5148700000376665 0.04803800000000001
  rotation 0.9999999985357193 3.826592070007306e-05 3.826592070014007e-05 1.5707963282591773
  name "wooden box(1)"
  size 0.1 0.1 0.1
  mass 5
}
RobocupSoccerBall {
  translation -2.38144 -0.763434 0.0736689
  rotation 0.680650427679195 0.2767804799866798 0.6783122888452426 -2.6014738031741977
}
Dog {
  translation -2.24658 -3.34026 0
  rotation 0 0 -1 -1.5708053071795867
  scale 0.4
}
Solid {
  translation -3.27 -2.05 0.05
  children [
    DEF red_ball Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
          emissiveColor 1 0 0
        }
      }
      geometry Sphere {
        radius 0.075
      }
    }
  ]
  name "solid_ball_red"
  boundingObject USE red_ball
}
Solid {
  translation -3.27 -2.35 0.05
  children [
    DEF blue_ball Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 1
          emissiveColor 0 0 1
        }
      }
      geometry Sphere {
        radius 0.075
      }
    }
  ]
  name "solid_ball_blue"
  boundingObject USE blue_ball
}
Solid {
  translation -3.27 -2.64 0.05
  children [
    DEF green_ball Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 1 0
          emissiveColor 0 1 0
        }
      }
      geometry Sphere {
        radius 0.075
      }
    }
  ]
  name "solid_ball_green"
  boundingObject USE green_ball
}
Solid {
  translation -3.27 -1.77 0.05
  children [
    DEF black_ball Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 0
        }
      }
      geometry Sphere {
        radius 0.075
      }
    }
  ]
  name "solid_ball_black"
  boundingObject USE black_ball
}
RobocupGoal {
  translation -1.29 -0.7 0
  rotation 0 0 1 3.141592653589793
  frameColor 1 0 0
  netColor 0 0 0
}
Solid {
  translation 2.19 -0.7797572032675383 0.03839095881449435
  rotation 1 0 0 1.5707963267948966
  children [
    DEF chap HingeJoint {
      endPoint Solid {
        translation 0 0.0015944916199695448 -0.07498304872752175
        rotation 1 0 0 0.021261490108714268
        children [
          DEF rast_cyl Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0 0 0
              }
            }
            geometry Cylinder {
              height 0.03
              radius 0.04
            }
          }
        ]
        name "solid(1)"
        boundingObject USE rast_cyl
        physics Physics {
        }
      }
    }
    DEF rast HingeJoint {
      endPoint Solid {
        translation 0 0.001763853235509012 0.07507928357252468
        rotation -1 0 0 0.033246693575794
        children [
          DEF rast_cyl Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0 0 0
              }
            }
            geometry Cylinder {
              height 0.03
              radius 0.04
            }
          }
        ]
        boundingObject USE rast_cyl
        physics Physics {
        }
      }
    }
    DEF mile_vasat Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 1 0
          emissiveColor 0 1 0
        }
      }
      geometry Cylinder {
        height 0.12
        radius 0.017
      }
    }
  ]
  name "weight"
  boundingObject USE mile_vasat
  physics Physics {
  }
}
Solid {
  translation 1.99 -0.85 0.0447739776
  rotation 0.2875560793353433 0.2984007083871668 -0.9100925878564899 1.0260849076917671e-16
  children [
    DEF le_st Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0.333333 1 1
        }
      }
      geometry Box {
        size 0.08 0.08 0.09
      }
    }
  ]
  name "left_stand"
  boundingObject USE le_st
  physics Physics {
  }
}
Solid {
  translation 1.99 -0.72 0.04477397760000001
  rotation 0.02256367606603789 0.09234455990011718 0.995471427404745 3.3163631048927527e-16
  children [
    DEF le_st Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0.333333 1 1
        }
      }
      geometry Box {
        size 0.08 0.08 0.09
      }
    }
  ]
  name "right_stand"
  boundingObject USE le_st
  physics Physics {
  }
}
